common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -41,34 +44,33 @@
    * @param buffer the byte buffer to be unmapped
    */
   public static void cleanDirectBuffer(ByteBuffer buffer) {
-    if (buffer == null) {
-      return;
-    }
-    if (buffer.isDirect()) {
-      try {
-        if (sCleanerMethod == null) {
-          sCleanerMethod = buffer.getClass().getMethod("cleaner");
-          sCleanerMethod.setAccessible(true);
-        }
-        final Object cleaner = sCleanerMethod.invoke(buffer);
-        if (cleaner != null) {
-          if (sCleanMethod == null) {
-            sCleanMethod = cleaner.getClass().getMethod("clean");
-          }
-          sCleanMethod.invoke(cleaner);
-        }
-      } catch (Exception e) {
-        LOG.warn("Fail to unmap direct buffer due to ", e);
-        buffer = null;
+    Preconditions.checkNotNull(buffer);
+    Preconditions.checkArgument(buffer.isDirect(), "buffer isn't a DirectByteBuffer");
```

Comment:
Previously this would not throw an error, does this mean its required to pass in a direct bytebuffer now?